src/basic/terminal-util.c

```diff
@@ -2427,6 +2445,158 @@ int terminal_get_size_by_dsr(
         return r;
 }
 
+#define MAX_TERMINFO_LENGTH 64
+/* python -c 'print("".join(hex(ord(i))[2:] for i in "TN").upper())' */
+#define DCS_TERMINFO_Q ANSI_DCS "+q" "544E" ANSI_ST
+/* The answer is either 0+r… (invalid) or 1+r… (OK). */
+#define DCS_TERMINFO_R0 ANSI_DCS "0+r" "544E" ANSI_ST
+#define DCS_TERMINFO_R1 ANSI_DCS "1+r" "544E" "=" /* This is followed by Pt ST. */
+
+static int scan_terminfo_response(
+                const char *buf,
+                size_t size,
+                char **ret_name) {
+        int r;
+
+        assert(buf);
+        assert(ret_name);
+
+        /* Check if the have enough space for an answer */
+        if (size < STRLEN(DCS_TERMINFO_R0))
```

Comment:
hmm, I would have preferred if this used CONST_MIN(_R0, _R1 ANSI_ST), or use assert_cc() above to make it clear that we're checking for the shortest valid reply